content/docs/command-reference/exp/show.md

```diff
@@ -168,6 +168,11 @@ $ dvc exp show ...
   automatically. You can enable `dvc config plots.auto_open` to make this the
   default behavior.
 
+- `-f`, `--force` - force re-collection of experiments instead of loading from
+  exp cache. DVC caches `exp show` table data to improve performance of
```

Comment:
do we cache the workspace too?

Reply:
No, this only caches results from git commits (which are fixed), we don't cache the workspace (and always re-collect it)

Reply:
should we mention this (otherwise it feels like this option should be used often, while it's more like an internal option).

Also, can it happen that we have some temporary errors in the revisions - do we cache in those cases?

Reply:
Yes, we cache the errors (per git commit) right now, so you would need to use `-f` to force re-collection if you fix whatever is causing the temporary error.

This is discussed in the original PR, but the issue we have right now is that in DVC during collection we don't differentiate between errors that are recoverable (a dvc-tracked metrics file couldn't be fetched from a remote due to network issues) and errors that are not (a git-committed metrics file is invalid, and will always be invalid every time we try to collect this git commit)

Reply:
Maybe it would clarify to say that data is cached for completed experiments?

Reply:
How do we integrate it with VS Code? It feels it would be way safer to always use `-f` there?

Can we for now stop caching any experiments with any errors? So that we don't have to advertise `-f` as a regular options to use (if something not working) ...

Reply:
I updated the PR to document that we only cache completed experiments, and disabled error caching for now on the DVC side.

For reference, disabling error caching makes `dvc exp show -A --json` 7x slower on the vscode demo repo on my machine (7s without error caching vs 1s with error caching). But either way `exp show` is still much faster than 2.45.1 (33s prior to any of the caching+collection performance improvements)

Reply:
🤔 
In that case maybe it's better to cache errors by default and have vs-code use --force every 1-2minutes only? so the nominal performance is better, but error recovery still exist, even if a bit less timely?

Reply:
it depends on what we expect. Do we expect users to have a lot of broken experiments permanently in their workspace and a few commits that they see in the table? If the answer is yes, I would think backwards also - how should the interface look like in VS Code and CLI for these failed exps. E.g. can we make it explicit (write a message in a each row that `-f` is needed to attempt to collect it again?) - then it might be fine to even have the implementation that @pmrowla had already. If we expect this to be more or less rare (a lot of permanent errors that slow us down) - I would say let's just not cache them for now.

Reply:
Thanks for those clarifications @pmrowla  - I say let's merge this as is and try to get concrete examples of issues related to error-recovery for next steps, hopefully with concrete scenarios we can prioritize, and then revisit as part of that